share/commands/technique-migrate

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+#. "${BASEDIR}/../lib/api_call.sh"
+. "/opt/rudder/share/lib/api_call.sh"
+
+TECHNIQUES_DIRECTORY="${CONFIGURATION_DIRECTORY}/technique"
+TOKEN=""
+DISPLAY_COMMAND=false
+API_URL="https://localhost/rudder"
+
+while getopts "iIvdqct:n:" opt; do
+  case $opt in
+    i|I|v|d)
+      DISPLAY_COMMAND=true
+      ;;
+    t)
+      TOKEN="${OPTARG}"
+      ;;
+    n)
+      TECHNIQUE_NAME="${OPTARG}"
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+# to keep the argument as $1
+shift  $((OPTIND-1))
+
+if [ -z "${TECHNIQUE_NAME}" ]
+then
+  echo "Usage: rudder technique migrate [-iIvd] -t token -n technique_name [target_technique_version]"
+  exit 1
+fi
+
+TECHNIQUE_VERSION="$1"
+if [ -z "${TECHNIQUE_VERSION}" ]
+then
+  TECHNIQUE_VERSION="latest"
+fi
+
+if [ "${TECHNIQUE_VERSION}" == "latest" ]
+then
+# Find the newest technique version
+  TECHNIQUE_VERSION=$(complete_api_call "${API_URL}/api/latest/techniques" "${TOKEN}" "GET" "|jq -r '.data | .techniques[] | select( .name == \"${TECHNIQUE_NAME}\") | .versions[]' | tail -n1" "${DISPLAY_COMMAND}")
+fi
+
+echo "TECHNIQUE ${TECHNIQUE_NAME} will migrate to ${TECHNIQUE_VERSION}"
+echo ""
+
+# Find the directives using the given technique
+DIRECTIVES=$(complete_api_call "${API_URL}/api/latest/directives" "${TOKEN}" "GET" "| jq -r '.data.directives[] | select(.techniqueName==\"${TECHNIQUE_NAME}\") | .id'" "${DISPLAY_COMMAND} | sed 's/\"//g'")
+
+echo "Directives using the '${TECHNIQUE_NAME}' technique:"
+echo "${DIRECTIVES}"
+echo ""
+
+while IFS= read
+do
+  # Change the version parameter in the previously found directives (ie 1 directive = 1 technique)
+    echo "Treating $REPLY"
+    result=$(complete_api_call "${API_URL}/api/latest/directives/$REPLY" "${TOKEN}" "POST" "-d \"techniqueVersion=${TECHNIQUE_VERSION}\"" "${DISPLAY_COMMAND}")
+    code=$?
+    if [ ${code} -eq 0 ]
+    then
+      printf "  ${GREEN}ok${NORMAL}: Changed technique version for directive: '$REPLY'.\n"
+    else
+      printf "  ${RED}error${NORMAL}: Could not change technique version for directive: '$REPLY'.\n"
+      echo "  ${result}"
+      exit 1
+    fi
+done <<< "${DIRECTIVES}"
```

Comment:
Bashism, you should either specify /bin/bash as a shebang or use echo "${DIRECTIVES}" |